pkg/plugins/policies/meshtimeout/api/v1alpha1/schema.yaml

```diff
@@ -0,0 +1,148 @@
+properties:
+  type:
+    description: ''
+    type: string
+    enum:
+      - MeshTimeout
+  mesh:
+    description: 'Mesh is the name of the Kuma mesh this resource belongs to. It may be omitted for cluster-scoped resources.'
+    type: string
+  name:
+    description: 'Name of the Kuma resource'
+    type: string
+  from:
+    description: From list makes a match between clients and corresponding configurations
+    items:
+      properties:
+        default:
+          description: Default is a configuration specific to the group of clients referenced in 'targetRef'
+          properties:
+            connectionTimeout:
+              description: ConnectionTimeout specifies the amount of time proxy will wait for an TCP connection to be established. Default value is 5 seconds. Cannot be set to 0.
+              type: object
+            http:
+              description: Http provides configuration for HTTP specific timeouts
+              properties:
+                maxConnectionDuration:
+                  description: if not set there is no max, cannot be 0 MaxConnectionDuration is the time after which a connection will be drained and/or closed, starting from when it was first established. Setting this timeout to 0 will disable it. Disabled by default.
+                  type: object
+                maxStreamDuration:
+                  description: MaxStreamDuration is the maximum time that a stream’s lifetime will span. Setting this timeout to 0 will disable it. Disabled by default.
+                  type: object
+                requestTimeout:
+                  description: set to 0 to disable, if not specified then disabled RequestTimeout The amount of time that proxy will wait for the entire request to be received. The timer is activated when the request is initiated, and is disarmed when the last byte of the request is sent, OR when the response is initiated. Setting this timeout to 0 will disable it. Default is 15s.
+                  type: object
+                streamIdleTimeout:
+                  description: StreamIdleTimeout is the amount of time that proxy will allow a stream to exist with no activity. Setting this timeout to 0 will disable it. Default is 30m
+                  type: object
+              type: object
+            idleTimeout:
+              description: IdleTimeout is defined as the period in which there are no bytes sent or received on connection Setting this timeout to 0 will disable it. Be cautious when disabling it because it can lead to connection leaking. Default value is 1h.
+              type: object
```

Comment:
How come these are objects? To me they should be strings no?

Reply:
https://github.com/kumahq/kuma/blob/978f0019959341e265e2a560570bfbf2814417b5/pkg/config/types/duration.go#L10-L12 - it's a `Duration` struct with custom parsing I'm guessing.

Reply:
Yes but the user API is "string" because you put things like "1s" there's likely a problem in the openapi spec generation here.

Reply:
@lobkovilya do you know what's up there?

Reply:
I think we should use k8s.io/apimachinery/pkg/apis/meta/v1.Duration here. It has the right method to generate the right openAPI schema

Reply:
changed as you Charly suggested to k8s duration